src/coreclr/src/jit/gentree.h

```diff
@@ -3108,12 +3123,193 @@ struct GenTreeLclVarCommon : public GenTreeUnOp
+
+    // Clear anything that was already there by masking out the bits before 'or'ing in what we want there.
+    newFlags = (unsigned char)((newFlags & ~(packedFlags << (idx * 2))) | (bits << (idx * 2)));
+    return newFlags;
+}
+
 // gtLclVar -- load/store/addr of local variable
 
 struct GenTreeLclVar : public GenTreeLclVarCommon
 {
+private:
+    regNumberSmall     gtOtherReg[MAX_MULTIREG_COUNT - 1];
+    MultiRegSpillFlags gtSpillFlags;
+    unsigned int GetLastUseBit(int regIndex)
+    {
+        assert(regIndex < 4);
+        return (1 << (MULTIREG_LAST_USE_SHIFT + regIndex));
+    }
+
+public:
     INDEBUG(IL_OFFSET gtLclILoffs;) // instr offset of ref (only for JIT dumps)
 
+    // Multireg support
+    bool IsMultiReg() const
+    {
+        return ((gtFlags & GTF_VAR_MULTIREG) != 0);
+    }
+    void ClearMultiReg()
+    {
+        gtFlags &= ~GTF_VAR_MULTIREG;
+    }
+    void SetMultiReg()
+    {
+        gtFlags |= GTF_VAR_MULTIREG;
+    }
+
+    regNumber GetRegNumByIdx(int regIndex)
+    {
+        assert(regIndex < MAX_MULTIREG_COUNT);
+        return (regIndex == 0) ? GetRegNum() : (regNumber)gtOtherReg[regIndex - 1];
+    }
+
+    void SetRegNumByIdx(regNumber reg, int regIndex)
+    {
+        assert(regIndex < MAX_MULTIREG_COUNT);
+        if (regIndex == 0)
+        {
+            SetRegNum(reg);
+        }
+        else
+        {
+            gtOtherReg[regIndex - 1] = regNumberSmall(reg);
+        }
+    }
+
+    bool IsLastUse(int regIndex)
+    {
+        return (gtFlags & GetLastUseBit(regIndex)) != 0;
+    }
+
+    bool HasLastUse()
+    {
+        return (gtFlags & (GTF_VAR_DEATH_MASK)) != 0;
+    }
+
+    void SetLastUse(int regIndex)
+    {
+        unsigned int bitToSet = gtFlags |= GetLastUseBit(regIndex);
+    }
+
+    void ClearLastUse(int regIndex)
+    {
+        gtFlags &= ~GetLastUseBit(regIndex);
+    }
+
+    unsigned GetRegSpillFlagByIdx(unsigned idx) const
+    {
+        return GetMultiRegSpillFlagsByIdx(gtSpillFlags, idx);
+    }
+
+    void SetRegSpillFlagByIdx(unsigned flags, unsigned idx)
+    {
+        gtSpillFlags = SetMultiRegSpillFlagsByIdx(gtSpillFlags, flags, idx);
+    }
+
+    unsigned int GetFieldCount(Compiler* compiler);
+    var_types GetFieldTypeByIndex(Compiler* compiler, unsigned idx);
+
+    //-------------------------------------------------------------------
+    // clearOtherRegFlags: clear GTF_* flags associated with gtOtherRegs
+    //
+    // Arguments:
+    //     None
+    //
+    // Return Value:
+    //     None
+    void ClearOtherRegFlags()
+    {
+        gtSpillFlags = 0;
+    }
+
+    //-------------------------------------------------------------------------
+    // CopyOtherRegFlags: copy GTF_* flags associated with gtOtherRegs from
+    // the given call node.
+    //
+    // Arguments:
+    //    fromCall  -  GenTreeCall node from which to copy
+    //
+    // Return Value:
+    //    None
+    //
+    void CopyOtherRegFlags(GenTreeLclVar* from)
```

Comment:
Does it accept from as lclVar or as a call? The header is probably incorrect.

Reply:
Ooops, yes, cut and paste error.